eureka/src/main/java/com/linecorp/armeria/server/eureka/EurekaUpdatingListener.java

```diff
@@ -223,17 +223,19 @@ private InstanceInfo fillAndCreateNewInfo(InstanceInfo oldInfo, Server server) {
         } else {
             hostnameOrIpAddr = hostName;
         }
-        final String healthCheckUrl = healthCheckUrl(hostnameOrIpAddr, oldInfo.getHealthCheckUrl(), portWrapper,
+        final String healthCheckUrl = healthCheckUrl(hostnameOrIpAddr, oldInfo.getHealthCheckUrlPath(),
+                                                     oldInfo.getHealthCheckUrl(), portWrapper,
                                                      healthCheckService, SessionProtocol.HTTP);
         final String secureHealthCheckUrl =
-                healthCheckUrl(hostnameOrIpAddr, oldInfo.getSecureHealthCheckUrl(), securePortWrapper,
+                healthCheckUrl(hostnameOrIpAddr, oldInfo.getHealthCheckUrlPath(),
+                               oldInfo.getSecureHealthCheckUrl(), securePortWrapper,
                                healthCheckService, SessionProtocol.HTTPS);
 
         return new InstanceInfo(instanceId, appName, oldInfo.getAppGroupName(), hostName, ipAddr,
                                 vipAddress, secureVipAddress, portWrapper, securePortWrapper, InstanceStatus.UP,
-                                oldInfo.getHomePageUrl(), oldInfo.getStatusPageUrl(), healthCheckUrl,
-                                secureHealthCheckUrl, oldInfo.getDataCenterInfo(),
-                                oldInfo.getLeaseInfo(), oldInfo.getMetadata());
+                                oldInfo.getHomePageUrl(), oldInfo.getStatusPageUrl(),
+                                oldInfo.getHealthCheckUrlPath(), healthCheckUrl, secureHealthCheckUrl,
```

Comment:
What do you think of introducing two fields?
```java
private final InstanceInfo originalInstanceInfo;
@Nullable
private InstanceInfo filledInstanceInfo;
```
so that we can call the original constructor of the `InstanceInfo` and make the newly-added one package-private?

Reply:
I bet your suggestion will work.. lemme think about it and try

Reply:
got closer.. I'll make another comment